### ISSUE TITLE
Cop idea: Shadowing Methods

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12379]
## Description

As a corollary, avoid shadowing methods with local variables unless they are both equivalent.

https://rubystyle.guide/#no-shadowing

## Behavior

```ruby
class Foo
  attr_accessor :options

  # ok
  def initialize(options)
    self.options = options
    # both options and self.options are equivalent here
  end

  # bad
  def do_something(options = {})
    unless options[:when] == :later
      output(self.options[:message])
    end
  end

  # good
  def do_something(params = {})
    unless params[:when] == :later
      output(options[:message])
    end
  end
end
```
